### ISSUE TITLE
linkAccountToLDAP must either use role or account type

### ISSUE THREAD

**Issue by borisstoyanov** [apache/cloudstack#3677]
##### ISSUE TYPE

 * Bug Report

##### COMPONENT NAME

~~~
API
~~~

##### CLOUDSTACK VERSION

~~~
4.14 
~~~

##### SUMMARY

currently the API supports only linking to an account type it's not supporting roles. 
For example if a user is using dynamic roles and wants to link ldap account, instead of role he needs to pass the account type, then comes the question if multiple roles are assigned to the same account type, what role will be assigned to him. 

##### EXPECTED RESULTS

~~~
Either pass account type or role
~~~

##### ACTUAL RESULTS

~~~
Supports just type
~~~

**Comment by borisstoyanov** [comment]
ping @rafaelweingartner you probably are using this, can you comment on the expected and actual behaviour.

**Comment by DaanHoogland** [comment]
marking this as a feature request.

**Comment by DaanHoogland** [comment]
To be clear @borisstoyanov you want the API linkAccountToLdap to take a role instead of an accounttype, am I right? so this account must not existe yet or already have the same role assigned to it for the link to succeed am I right?

**Comment by DaanHoogland** [comment]
@rafaelweingartner any ideas? or @PaulAngus , maybe?

**Comment by sureshanaparti** [comment]
Fixed in #8236
